File: android/app/src/main/java/com/rbdarts/feature/practice/PracticeService.kt
package com.rbdarts.feature.practice

import com.rbdarts.core.domain.PracticeAttempt
import com.rbdarts.core.domain.PracticeStats
import com.rbdarts.core.domain.ScoringRules

class PracticeService {
    fun validate(target: Int, score: Int) {
        ScoringRules.validateInning(target)
        ScoringRules.validateScore(score)
    }

    fun stats(attempts: List<PracticeAttempt>): PracticeStats =
        ScoringRules.practiceStats(attempts)
}

File: ios/RBDarts/Features/Practice/PracticeService.swift
import Foundation

public struct PracticeService: Sendable {
    public init() {}

    public func validate(target: Int, score: Int) throws {
        try ScoringRules.validate(inningNumber: target)
        try ScoringRules.validate(score: score)
    }

    public func stats(for attempts: [PracticeAttempt]) -> PracticeStats {
        ScoringRules.practiceStats(from: attempts)
    }
}
